api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -22,13 +22,18 @@ public boolean eligibleToTakeSurvey(DbUser user) {
       return false;
     }
 
-    final Instant createdTime = user.getCreationTime().toInstant();
+    final Instant now = clock.instant();
+    return now.isAfter(eligibilityWindowStart(user)) && now.isBefore(eligibilityWindowEnd(user));
```

Comment:
just to save a few cycles, you can extract eligibilityWindowStart to a variable and pass it to eligibilityWindowEnd. To avoid calling eligibilityWindowStart twice

Reply:
I'm not sure how to do this and keep `eligibilityWindowEnd` as a public function for the API to use, could you show me an example?

If it helps, the call to `user.getCreationTime()` will only hit the DB once since hibernate caches the response.

Reply:
I didn't notice that eligibilityWindowEnd is a public function for the API. 
An idea is to have a method 
private Instant eligibilityWindowEnd(Instant start) that adds the 61 days and call it in your methods with different parameters.

In this line you'll extract start to a variable
then call eligibilityWindowEnd(start)

while in your API method you can change to eligibiltyWindowEnd(DbUser user) {
    return eligibilityWindowEnd(eligibilityWindowStart(user));
}

Reply:
I'll keep this in mind, and if I think of something better in the next couple days I'll update it!

For now, I prefer to keep the logic of "x days after creation date" together. Ideally, I'd like `NewUserSatisfactionSurveyService` to encapsulate all business logic for the survey.